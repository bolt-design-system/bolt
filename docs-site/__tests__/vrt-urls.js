const path = require('path');
const fs = require('fs-extra');
const prettier = require('prettier');
const { getConfig } = require('@bolt/build-utils/config-store');

async function getVrtUrlsPath() {
  const config = await getConfig();

  return path.resolve(
    path.join(config.wwwDir, 'build/data/vrt-urls.bolt.json'),
  );
}

describe('Bolt VRT URLs', () => {
  test('VRT URL file exists and matches snapshot', async () => {
    const vrtUrlsPath = await getVrtUrlsPath();

    expect(
      fs.existsSync(vrtUrlsPath),
      'You may not have built Pattern Lab yet. Try running `yarn start` or `yarn build` to generate the missing file.',
    ).toBe(true);

    const results = fs.readFileSync(vrtUrlsPath, 'utf8');
    const formattedJSON = prettier.format(results, {
      parser: 'json',
    });

    expect(
      formattedJSON,
      'If changes are expected, run `yarn test:vrt-urls -u` from the project root to update snapshot. Then, inform the QA team of these changes.',
    ).toMatchSnapshot();
  });
});
