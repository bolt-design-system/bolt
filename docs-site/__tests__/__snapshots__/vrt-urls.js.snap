// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bolt VRT URLs VRT URL file exists and matches snapshot 1`] = `
"{
  \\"components\\": [
    \\"https://boltdesignsystem.com/pattern-lab/patterns/20-elements-button/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/20-elements-text-link/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-accordion/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-action-blocks/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-background/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-backgrounds-shapes/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-band/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-banner/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-block-list/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-blockquote/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-breadcrumb/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-button/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-card-replacement/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-carousel/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-chip/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-chips-list/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-code-snippet/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-copy-to-clipboard/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-device-viewer/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-figure/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-form/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-headline/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-hero/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-icon/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-image/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-link/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-list/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-logo/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-menu/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-modal/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-nav-priority/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-navbar/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-ol/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-pagination/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-popover/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-post/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-progress-bar/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-ratio/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-share/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-stack/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-sticky/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-table/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-tabs/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-teaser/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-text/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-toc/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-tooltip/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-trigger/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-typeahead/index.html\\",
    \\"https://boltdesignsystem.com/pattern-lab/patterns/40-components-ul/index.html\\"
  ]
}
"
`;
